models/intermediate/int__ip_visits.sql

```diff
@@ -1,84 +1,107 @@
+            PARTITION BY patient_id
+            ORDER BY event_date, event_type
+        ) AS overall_ord
     FROM (
         SELECT
             patient_id
-            , encounter_class
-            , event_datetime
-            , event_type
-            , max(start_ordinal)
-                OVER (
-                    PARTITION BY
-                        patient_id
-                        , encounter_class
-                    ORDER BY event_datetime, event_type ROWS UNBOUNDED PRECEDING
-                )
-            AS start_ordinal
+            , encounter_start_date AS event_date
+            , -1 AS event_type
             , row_number() OVER (
-                PARTITION BY patient_id, encounter_class
-                ORDER BY event_datetime, event_type
-            ) AS overall_ord
-        FROM (
-            SELECT
-                patient_id
-                , encounter_class
-                , encounter_start_datetime AS event_datetime
-                , -1 AS event_type
-                , row_number() OVER (
-                    PARTITION BY patient_id, encounter_class
-                    ORDER BY encounter_start_datetime, encounter_stop_datetime
-                ) AS start_ordinal
-            FROM {{ ref( 'stg_synthea__encounters') }}
-            WHERE encounter_class = 'inpatient'
-            UNION ALL
-            SELECT
-                patient_id
-                , encounter_class
-                , {{ dbt.dateadd(datepart="day", interval=1, from_date_or_timestamp="encounter_stop_datetime") }} AS event_datetime
-                , 1 AS event_type
-                , NULL AS start_ordinal
-            FROM {{ ref( 'stg_synthea__encounters') }}
-            WHERE encounter_class = 'inpatient'
-        ) AS rawdata
+                PARTITION BY patient_id
+                ORDER BY encounter_start_date, encounter_stop_date
+            ) AS start_ordinal
+        FROM all_ip_encounters
+        UNION ALL
+        SELECT
+            patient_id
+            , {{ dbt.dateadd(datepart="day", interval=1, from_date_or_timestamp="encounter_stop_date") }} AS event_date
+            , 1 AS event_type
+            , NULL AS start_ordinal
+        FROM all_ip_encounters
     ) AS e
-    WHERE (2 * e.start_ordinal - e.overall_ord = 0)
+)
+
+, cte_end_dates AS (
+    SELECT
+        patient_id
+        , {{ dbt.dateadd(datepart="day", interval=-1, from_date_or_timestamp="event_date") }} AS end_date
+    FROM cte_collapse_encounters
+    WHERE (2 * start_ordinal - overall_ord = 0)
 )
 
 , cte_visit_ends AS (
     SELECT
-        min(v.encounter_id) AS encounter_id
-        , v.patient_id
-        , v.encounter_class
-        , v.encounter_start_datetime AS visit_start_datetime
-        , min(e.end_datetime) AS visit_end_datetime
-    FROM {{ ref( 'stg_synthea__encounters') }} AS v
+        min(a.encounter_id) OVER (PARTITION BY a.patient_id, a.encounter_start_date) AS encounter_id
+        , a.encounter_id AS original_encounter_id
```

Comment:
most of the code in this model is unchanged (rows just got shifted around). this line is an exception. i'm retaining the original encounter ID so we can map all the rolled up encounters during an IP stay back to the visit occurrence for that stay.